### PR TITLE
KTOR-1858 Fix URL double encoding in Darwin

### PR DESCRIPTION
**Pull request by rsinukov**

### CODE REVIEW

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpRedirectTest.kt [hunk 1]
```diff
@@ -39,7 +39,7 @@ class HttpRedirectTest : ClientLoader() {
         test { client ->
             client.prepareGet("$TEST_URL_BASE/encodedQuery").execute {
                 assertEquals(HttpStatusCode.OK, it.status)
-                assertEquals("/redirect/getWithUri?key=value1%3Bvalue2%3D%22some=thing", it.bodyAsText())
+                assertEquals("/redirect/getWithUri?key=value1%3Bvalue2%3D%22some%3Dthing", it.bodyAsText())
```
Comment (e5l) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpRedirectTest.kt:42]: please consider reverting this change: it's a major breaking change

**Review by e5l — CHANGES_REQUESTED** [review]

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpRedirectTest.kt (hunk 1 above)
Comment (rsinukov) [reply ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpRedirectTest.kt]: done

File: ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/internal/DarwinUrlUtils.kt [hunk 2]
```diff
@@ -5,28 +5,76 @@
...
+            .map { NSURLQueryItem(it.first.encodeQueryKey(), it.second?.encodeQueryValue()) }
+    }
+
+    components.percentEncodedFragment = when {
+        encodedFragment.isEmpty() -> null
+        fragmentEncoded -> encodedFragment
+        else -> fragment.sanitize(NSCharacterSet.URLFragmentAllowedCharacterSet)
+    }
+
+    return components.URL!!
```
Comment (e5l) [inline ktor-client/ktor-client-darwin/darwin/src/io/ktor/client/engine/darwin/internal/DarwinUrlUtils.kt:62]: we can add something like `?: error("Invalid url: $this") to make it easier for debug

**Review by e5l — APPROVED** [review]
lgmt

**Pull request merged**